File: tests/fixtures/python/factories.py
from repos import AuditLog, UserRepository
from store import connect


def make_repo() -> UserRepository:
    return UserRepository()


def make_audit():
    return AuditLog()


def rotate(user_id: int) -> None:
    repo = make_repo()
    repo.delete_user(user_id)
    audit = make_audit()
    audit.delete_user(user_id)
    session = connect()
    session.close()


def forget(repo, user_id: int):
    repo.find_user(user_id)
    return repo.delete_user(user_id)

File: tests/fixtures/typescript/factories.ts
import { AuditLog, UserRepository } from "./repos";
import connect from "./store";

export function makeRepo(): UserRepository {
  return new UserRepository();
}

export function makeAudit() {
  return new AuditLog();
}

export async function rotate(id: number): Promise<void> {
  const repo = makeRepo();
  await repo.deleteUser(id);
  const audit = makeAudit();
  audit.deleteUser(id);
  const session = connect();
  session.close();
}

export function forget(repo: any, id: number): void {
  repo.findUser(id);
  void repo.deleteUser(id);
}
